Updated tracestate to specify that it should not be used for application level properties

kalyanaj opened the pull request:
Updated tracestate section to specify that it should not be used for application level properties. Resolves #439.


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/kalyanaj/trace-context/pull/458.html" title="Last updated on Aug 31, 2021, 7:23 PM UTC (f647fb3)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/trace-context/458/0d32b21...kalyanaj:f647fb3.html" title="Last updated on Aug 31, 2021, 7:23 PM UTC (f647fb3)">Diff</a>

kalyanaj committed:
Updated tracestate section to specify that it should not be used for application level properties.

SergeyKanzhelev reviewed (changes requested):
specs links must be done differently. Also, normative section cannot refer to WD documents

SergeyKanzhelev reviewed (changes requested):
> The `tracestate` HTTP header MUST NOT be used for _application_ level properties such as userId."

My problem is that without the link to spec, that explains more about what "application level properties" are, we need more explanaition in the spec. Why `trace-id` is not application level for instance? Is `application-id` or "cluster" an applicaton level property? I know some vendors want to propagate the "cluster" or "application-id" which help to route traces to the right trace backend "bucket"

plehegar commented on spec/20-http_request_header_format.md:
```diff
@@ -227,6 +227,8 @@ The main purpose of the `tracestate` HTTP header is to provide additional vendor
 
 If the vendor failed to parse `traceparent`, it MUST NOT attempt to parse `tracestate`. Note that the opposite is not true: failure to parse `tracestate` MUST NOT affect the parsing of `traceparent`.
 
+The `tracestate` HTTP header MUST NOT be used for _application_ level properties such as userId. [W3C baggage](https://github.com/w3c/baggage) provides the specification for defining and propagating application level properties.
```
```suggestion
The `tracestate` HTTP header MUST NOT be used for _application_ level properties such as userId. [[BAGGAGE]] MAY be used for defining and propagating application level properties.
```

plehegar replied:
Note that the reference will work once Baggage is published (this Thursday)

kalyanaj committed:
Update spec/20-http_request_header_format.md

Committing suggestion to add the "MAY" semantics and the right way to reference the baggage spec.

Co-authored-by: Philippe Le Hegaret <plh@w3.org>

kalyanaj committed:
Clarifying scope of tracestate in the context of application level properties based on PR feedback.

dyladan commented on spec/20-http_request_header_format.md:
```diff
@@ -227,6 +227,8 @@ The main purpose of the `tracestate` HTTP header is to provide additional vendor
 
 If the vendor failed to parse `traceparent`, it MUST NOT attempt to parse `tracestate`. Note that the opposite is not true: failure to parse `tracestate` MUST NOT affect the parsing of `traceparent`.
 
+The `tracestate` HTTP header MUST NOT be used for any properties that are not defined by a tracing vendor. [[BAGGAGE]] MAY be used for defining and propagating such application level properties.
```
We should avoid the term "vendor" where possible as we may be used by open source tools too. I would use the word "telemetry system" or similar

kalyanaj committed:
Incorporate PR feedback/comments.

SergeyKanzhelev commented on spec/20-http_request_header_format.md:
```diff
@@ -227,6 +227,8 @@ The main purpose of the `tracestate` HTTP header is to provide additional vendor
 
 If the vendor failed to parse `traceparent`, it MUST NOT attempt to parse `tracestate`. Note that the opposite is not true: failure to parse `tracestate` MUST NOT affect the parsing of `traceparent`.
 
+The `tracestate` HTTP header MUST NOT be used for any properties that are not defined by a telemetry system. [[BAGGAGE]] MAY be used for defining and propagating such application level properties.
```
```suggestion
The `tracestate` HTTP header MUST NOT be used for any properties that are not defined by a tracing system. [[BAGGAGE]] MAY be used for defining and propagating such application level properties.
```

We call it "tracing systems"

kalyanaj committed:
Update spec/20-http_request_header_format.md

Changed terminology from telemetry system to tracing system per PR feedback

Co-authored-by: Sergey Kanzhelev <S.Kanzhelev@live.com>
